geode-core/src/main/java/org/apache/geode/management/internal/api/LocatorClusterManagementService.java

```diff
@@ -256,7 +256,9 @@ public ClusterManagementResult update(CacheElement config) {
       CacheConfig currentPersistedConfig = persistenceService.getCacheConfig(group, true);
       List<? extends T> listInGroup = manager.list(filter, currentPersistedConfig);
       for (T element : listInGroup) {
-        if (group.equals(element.getConfigGroup()) || element instanceof MultiGroupCacheElement) {
+        if (filter.getGroup() == null || // if listing all groups
+            group.equals(filter.getGroup()) || // if filter group matches this group
+            element instanceof MultiGroupCacheElement) { // if element can span multi groups
```

Comment:
`element instanceof MultiGroupCacheElement` feels incorrect to me.  I think we should be adding the result to the list iff:
1) no group filter was specified, OR
2) group filter matches the element's group, OR
3) the element type is cluster-level only (i.e. *not* instanceof MultiGroupCacheElement)

So I think the 3rd condition should be `!(element instanceof MultiGroupCacheElement)`

Reply:
or is the desired behavior really that listing cluster-level elements with a group filter specified should always return an empty list?  maybe if would be better to throw exception saying that group filter is not allowed?